### DISCUSSION TITLE
Support for p.connect parameters in pybullet_utils' BulletClient

### DISCUSSION THREAD

**Discussion by maximilian-maurer** [bulletphysics/bullet3#3675] [General]
The `pybullet_utils.bullet_client.BulletClient` wrapper is very useful when running many pybullet simulations at once. However, it only supports construction with a `connection_mode` and it is not possible to supply additional `kwargs` to it to pass onto the `pybullet.connect` call, such as the shared memory key or TCP/UDP addresses. 

While it is trivial to work around this by duplicating the class locally and adding this, or by just not using it and supplying the physicsClientId manually to function calls, it would be a nice feature to have for it to be able to pass these parameters through. It is a very small thing, but if it were added it would round-out the class for this use-case which does not seem to be covered. (Or am I just missing it and it is actually already possible?)

**Comment by ManifoldFR** [comment]
Would something like [this](https://github.com/facebookresearch/ScaDiver/blob/master/bullet/bullet_client.py) from a recent Facebook paper work ?
